### PR TITLE
Koa2

### PR DESCRIPTION
**Pull request by coderhaoxin**

### CODE REVIEW

**Comment by coderhaoxin** [comment]
It seems that `http://isaacs.couchone.com/registry/_all_docs?limit=5` is inaccessible.

**Comment by kilianc** [comment]
oh thanks, I'll look for some other public url that's available

**Comment by coderhaoxin** [comment]
http://httpbin.org/

File: lib/jsonp.js [hunk 1]
```diff
@@ -17,36 +17,37 @@ module.exports = function jsonp(options) {
...
-        endChunk: endChunk
-      }))
-    } else {
-      this.body =  startChunk + JSON.stringify(this.body, null, this.app.jsonSpaces) + endChunk
-
-      // JSON parse vs eval fix. https://github.com/rack/rack-contrib/pull/37
-      this.body = this.body.replace(/\u2028/g, '\\u2028').replace(/\u2029/g, '\\u2029')
-    }
+  return function (ctx, next) {
+    return next().then(() => {
```
Comment (kilianc) [inline lib/jsonp.js]: so we can't yield promises anymore ?

Comment (coderhaoxin) [inline lib/jsonp.js]: > can't yield promises anymore

Yes!

But we can use async/await (transplate to promise by `babel` for now, which is my own preference.)

```js
return async function (ctx, next) {
  await next()
  // ...
```

Comment (kilianc) [inline lib/jsonp.js]: bullshit :D

**Comment by coderhaoxin** [comment]
@kilianc since koa-jsonp is still pre release version(0.x).
Maybe it's better to bump deps and publish v1 for koa@1, then published v2 for koa@2.
And set latest on npm to v1(for koa@1), add a dist-tag next for koa-jsonp@2

**Comment by coderhaoxin** [comment]
All tests passed now!

**Comment by kilianc** [comment]
@coderhaoxin I was thinking about having `@next` as fixed tag for this and have a separate branch called `koa@2`. This requires a manual manual pull and I am out for node interactive, aka unable to sit down and work until tomorrow.

Talk to you soon

**Comment by coderhaoxin** [comment]
NVM >_<

Just add #6 for koa@1(master), and this PR is based on #6

**Comment by kilianc** [comment]
Thanks I will be clearing this up tomorrow. :beers:

**Comment by coderhaoxin** [comment]
@kilianc I changed `promise` to `async/await`, it's almost 1:1 change. And we just use babel to compile util node(v8) land async/await

**Comment by kilianc** [comment]
This is great! pulling this now

**Comment by coderhaoxin** [comment]
thx

**Comment by kilianc** [comment]
merged in https://github.com/kilianc/koa-jsonp/tree/koa%402 and published on npm with `npm i koa-json-@next`.

Whoever decided to drop `co` for `async` in `koa` made a mistake. `async` is not even close to be as powerful as `co`. No flow control, no thunks, no generators. It should be smart enough, if you pass `async function` doesn't use co, if you pass `function *` it uses co.

I am not going to start transpiling my code for a downgrade in features :)

**Pull request closed without merging**